src/core/api_client_parallel.py

```
@@ -37,80 +36,98 @@ async def _get_with_key(self, session: aiohttp.ClientSession, key_index: int,
             tasks = []
             for i, (endpoint, params) in enumerate(requests):
                 key_index = i % self.num_keys  # Distribute across keys
                 task = self._get_with_key(session, key_index, endpoint, params)
                 tasks.append(task)
+
+            # Execute all tasks in parallel with limited concurrency
+            semaphore = asyncio.Semaphore(self.num_keys * 2)
             
-            # Execute all tasks in parallel
-            responses = await asyncio.gather(*tasks, return_exceptions=True)
+            async def bounded_task(task):
```

Ruff (0.12.2)

Missing return type annotation for private function `bounded_task`

(ANN202)